src/lib/plot/bar.typ

```diff
@@ -52,19 +52,30 @@
     let right = left + w
     let width = (right - left) / len
 
+
     if self.mode in ("basic", "clustered") {
       left = left + width * n
       right = left + width
     }
-
     if (left <= x-axis.max and right >= x-axis.min and
         y-min <= y-axis.max and y-max >= y-axis.min) {
       left = calc.max(left, x-axis.min)
       right = calc.min(right, x-axis.max)
       y-min = calc.max(y-min, y-axis.min)
       y-max = calc.min(y-max, y-axis.max)
 
-      draw.rect((left, y-min), (right, y-max))
+
+      draw.group(ctx => {
+        draw.rect((left, y-min), (right, y-max))
+        if ctx.style.data-label != none {
+          let offset = ctx.style.data-label.at("offset")
+          let size = ctx.style.data-label.at("text-size")
+          let data_label = text(size:size)[#y-max]
+          let anchor = if y-axis.horizontal {"west"} else {"south"}
+          draw.content((rel: (0, offset), to: ((left + right) / 2, y-max)), anchor:"south", data_label)
```

Comment:
Prefer giving a name to the bar rect and using the `"north"/"south"` or `"east"/"west"` anchors, depending on the orientation _and_ value.

Offset must work for horizontal and vertical bars!